Improve Error message for `aiida_local_code_factory` fixture

So far when the executable which I want to use in the `aiida_local_code_factory` fixture is not found in the PATH the error message does not tell this directly:
```
aiida_local_code_factory = <function aiida_local_code_factory.<locals>.get_code at 0x11483e5f0>

    @pytest.fixture(scope='function')
    def voronoi_code(aiida_local_code_factory):
        """Get a voronoi code.
        """
>       VoroCode = aiida_local_code_factory(executable='voronoi.exe', entry_point='kkr.voro')

conftest.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../venv_aiida_fresh_install_py3/lib/python3.7/site-packages/aiida/manage/tests/pytest_fixtures.py:139: in get_code
    remote_computer_exec=[computer, executable_path],
../../../venv_aiida_fresh_install_py3/lib/python3.7/site-packages/aiida/orm/nodes/data/code.py:46: in __init__
    self.set_remote_computer_exec(remote_computer_exec)
../../../venv_aiida_fresh_install_py3/lib/python3.7/site-packages/aiida/orm/nodes/data/code.py:364: in set_remote_computer_exec
    if not os.path.isabs(remote_exec_path):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

s = None

    def isabs(s):
        """Test whether a path is absolute"""
>       s = os.fspath(s)
E       TypeError: expected str, bytes or os.PathLike object, not NoneType
```

It is probably better to give a hint in the error message what might be the reason for the `TypeError`.

--- a/aiida/manage/tests/pytest_fixtures.py
+++ b/aiida/manage/tests/pytest_fixtures.py
@@ -134,6 +134,9 @@ def aiida_local_code_factory(aiida_localhost):  # pylint: disable=redefined-oute
 
         executable_path = shutil.which(executable)
 
+        if not executable_path:
+            raise ValueError('The executable "{}" was not found in the $PATH.'.format(executable))
+
         code = Code(
             input_plugin_name=entry_point,
             remote_computer_exec=[computer, executable_path],

